### PR TITLE
Order primary branches at the top so that 'master' appears at the top…

### PR DESCRIPTION
**Pull request by i386**
… of branches tab, branch filter, etc ahead of favorites

To test go to the branch tab or branch filter and see that `master` is at the top.

# Description

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by i386** [comment]
@vivek quick one for you :)

**Comment by vivek** [comment]
LGTM 🐝

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/BranchImpl.java [hunk 1]
```diff
@@ -62,9 +64,6 @@ public PullRequest getPullRequest() {
     public Branch getBranch() {
         ObjectMetadataAction om = job.getAction(ObjectMetadataAction.class);
         PrimaryInstanceMetadataAction pima = job.getAction(PrimaryInstanceMetadataAction.class);
-        if (om == null && pima == null) {
```
Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/BranchImpl.java]: @vivek a `BranchInfo` is always a branch and should return a branch metadata

**Comment by i386** [comment]
@vivek all these tests now pass.

**Comment by vivek** [comment]
@i386 great, looks good.

**Pull request merged**